pkg/apis/enterprise/v1/monitoringconsole_types.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2018-2021 Splunk Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	splcommon "github.com/splunk/splunk-operator/pkg/splunk/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// default all fields to being optional
+// +kubebuilder:validation:Optional
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+// see also https://book.kubebuilder.io/reference/markers/crd.html
+
+// MonitoringConsoleSpec defines the desired state of MonitoringConsole
+type MonitoringConsoleSpec struct {
+	CommonSplunkSpec `json:",inline"`
+
+	// Splunk Smartstore configuration. Refer to indexes.conf.spec and server.conf.spec on docs.splunk.com
+	SmartStore SmartStoreSpec `json:"smartstore,omitempty"`
+}
+
+// MonitoringConsoleStatus defines the observed state of MonitoringConsole
+type MonitoringConsoleStatus struct {
+	// current phase of the monitoring console
+	Phase splcommon.Phase `json:"phase"`
+
+	// selector for pods, used by HorizontalPodAutoscaler
+	Selector string `json:"selector"`
+
+	// Splunk Smartstore configuration. Refer to indexes.conf.spec and server.conf.spec on docs.splunk.com
+	SmartStore SmartStoreSpec `json:"smartstore,omitempty"`
```

Comment:
@sgontla do we need smartstore settings for MC?

Reply:
No, we don't need it. Same for the line 50